pkg/collector/osmlogs_collector.go

```diff
@@ -0,0 +1,136 @@
+package collector
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// OsmLogsCollector defines an OsmLogs Collector struct
+type OsmLogsCollector struct {
+	BaseCollector
+}
+
+var _ interfaces.Collector = &OsmLogsCollector{}
+
+// NewOsmLogsCollector is a constructor
+func NewOsmLogsCollector(exporter interfaces.Exporter) *OsmLogsCollector {
+	return &OsmLogsCollector{
+		BaseCollector: BaseCollector{
+			collectorType: OsmLogs,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *OsmLogsCollector) Collect() error {
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	// ** Collect ground truth on OSM resources **
+	var groundTruthMap = map[string][]string{
+		"allResourcesTable":               []string{"get", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=wide"},
+		"allResourcesConfigs":             []string{"get", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
+		"mutatingWebhookConfigurations":   []string{"get", "MutatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
+		"validatingWebhookConfigurations": []string{"get", "ValidatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
+	}
+
+	for fileName, kubeCmd := range groundTruthMap {
+		if err = collectResourceToFile(collector, rootPath, fileName, kubeCmd); err != nil {
+			fmt.Printf("Failed to collect %s for OSM: %+v", fileName, err)
+		}
+	}
+
+	// ** Collect information for various resources across all meshes in the cluster **
+	meshList, err := getResourceList("deployments", "app=osm-controller", "-o=jsonpath={..meshName}", " ")
+	if err != nil {
+		return err
+	}
+
+	for _, meshName := range meshList {
+		namespacesInMesh, err := getResourceList("namespaces", "openservicemesh.io/monitored-by="+meshName, "-o=jsonpath={..name}", " ")
+		if err != nil {
+			return err
+		}
+		if err = collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName); err != nil {
+			fmt.Printf("Failed to collect data from OSM monitored namespaces: %+v", err)
+		}
+		if err = collectControllerLogs(collector, rootPath, meshName); err != nil {
+			fmt.Printf("Failed to collect OSM controller logs for mesh %s: %+v", meshName, err)
+		}
+	}
+
+	return nil
+}
```

Comment:
perhaps we introduce a new concept of a "meshCollector", or more generally something like a "chainableCollector", which is defined in the .yaml as a parent object under which the various collectors that act on the outputs of the "chainableCollector" are defined, along with whatever is needed to define the parent -> child relationship (templating?)